quickwit/quickwit-query/src/tokenizers/multilang.rs

```diff
@@ -0,0 +1,299 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use lindera_core::mode::Mode;
+use lindera_dictionary::{load_dictionary_from_config, DictionaryConfig, DictionaryKind};
+use lindera_tantivy::stream::LinderaTokenStream;
+use lindera_tantivy::tokenizer::LinderaTokenizer;
+use once_cell::sync::Lazy;
+use tantivy::tokenizer::{SimpleTokenStream, SimpleTokenizer};
+use tantivy_tokenizer_api::{Token, TokenStream, Tokenizer};
+use whichlang::{detect_language, Lang};
+
+// Use lazy lindera tokenizers to load the dictionaries only when needed.
```

Comment:
The comment is misleading. "When needed" today actually means on startup.
The reason why you used a lazy thing is for caching.

```suggestion
```